### PR TITLE
Show multiple cheat sheets when found (#147)

### PR DESCRIPTION
**Pull request by chubin**
If there are multiple cheat sheets with the same name
in several upstream repositories, show all of them.

To limit output with one single source only,
its name must be prepended to the query with `:`:

```
# show the dd cheat sheet from tldr only
curl cht.sh/tldr:dd

# show the dd cheat sheet from all repositories
curl cht.sh/dd
```

Selection_072

If there are no cheat sheets in the specified repository found,
the default fall back is used.

### CODE REVIEW

File: lib/routing.py [hunk 1]
```diff
@@ -111,7 +120,9 @@ def _get_page_dict(self, query, topic_type, request_options=None):
 
     def handle_if_random_request(self, topic):
         """
-        Check if the `query` if a :random one, if yes we check its correctness and then randomly select a topic, based on the provided prefix.
+        Check if the `query` if a :random one,
```
Comment (abitrolly) [inline lib/routing.py:123]:
```suggestion
        Check if the `query` is a :random one,
```

**Comment by abitrolly** [comment]
LGTM. Using f-strings an typing bumps the minimal Python to 3.6. I guess there no value in supporting Python 2.7 compatible codebase for this project.

I think that maybe some verbose mode should also show a direct link to the cheat source code.

**Review by abitrolly — APPROVED** [review]

**Comment by chubin** [comment]
Yes and yes.

I think one should switch to Python 3.6+ completely, and cradually update the whole codebase accordingly
(using f-strings, typing hints, etc.).

Regarding the direct link, I plan to add in the same format as we have for stackoverflow:

```
#  [Justin Peel] [so/q/2835559] [cc by-sa 3.0]
```

**Comment by abitrolly** [comment]
Makes sense for SO. Although Google fails to find [`[so/q/2835559]`](https://www.google.com/search?q=so%2Fq%2F2835559) - it can only make one's lucky with something like `stackoverflow:2835559`.

**Comment by chubin** [comment]
Actually, you are not supposed to use Google here. Just replace so with `stackoverflow.com`.
It must be a hyperlink in future (not all terminals support this, but many do)

**Comment by abitrolly** [comment]
The direct hyperlink to the cheatsheet is what I missed in the past. Discovering where do they come from takes time. That's why it is nice to have verbose mode with full source details.

**Pull request merged**